crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -973,11 +968,20 @@ impl IdentityManager {
             .await;
 
             // Once all the tasks have completed, process the results.
+            let mut updated_users = Vec::new();
             for result in results {
                 if let Some((user_id, updated_devices)) = result? {
                     devices_by_user.insert(user_id.to_owned(), updated_devices);
+                    updated_users.push(user_id);
                 }
             }
+
+            if !updated_users.is_empty() {
+                info!(
+                    ?updated_users,
+                    "Waited for `/keys/query` to complete for users who have no devices"
+                );
```

Comment:
Not sure if we could easily use `Itertools::format` here too, but I won't insist on that one.

Reply:
Yeah, I think this would be slightly trickier. We'd end up iterating `results` another *two* times: once to decide if we need to write the line at all, and again to format the log line:

```rust
            if results.iter().any(Option::is_some) {
                info!(
                    results.iter().filter_map(|r| r).map(|(user_id, _devices)| user_id).format(),
                    "Waited for `/keys/query` to complete for users who have no devices"
                );
            }
```